thunder/core/symbol.py

```diff
@@ -320,6 +321,18 @@ def __call__(self, *args, **kwargs):
             result = self.meta(*args, **kwargs)
             trace.pop_scope()
 
+        cd = get_compile_data()
+        if cd is not None and not cd.is_grad_enabled:
```

Comment:
There are paths in test which compile_data could be False - example while using `thunder.trace`

Reply:
Do we still have cases where thunder.trace is used without a compile data coming from elsewhere?

Reply:
These are the hits for `thunder.trace` with no compile-data (mainly in tests and some in example notebooks)

https://github.com/search?q=repo%3ALightning-AI%2Flightning-thunder%20thunder.trace()&type=code

Reply:
Right, to my mind, we should try to drop them. Thanks for bringing this up.